### DISCUSSION TITLE
I need text input  with locale label/image

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#9729] [Help] [answered]
In laravel 10 / filamentphp 3 I added akaunting/laravel-money package and can show money input
like : https://img001.prntscr.com/file/img001/KiUBZGV8SQ2CUzuB2LNzEA.png   

I need similar component with locale label/image.
How can I do it? Or maybe there such components ?

**Comment by ShaungBhone** [comment]
I don't get it.

**Comment by coolsam726** [comment, marked as answer]
I am not sure if this is what you want, but you can try something on these lines. It's up to you to decide how to determine the country or obtain the country's flag image. The point here is that you can use a closure that returns a HTML string to render your prefix or suffix.

```php
Forms\Components\TextInput::make('amount')->numeric()
->prefix(fn() => new HtmlString('<img src="https://flagsapi.com/KE/flat/32.png">')),
```

![image](https://github.com/filamentphp/filament/assets/5610289/345fe829-7688-47dc-825b-996616b3513e)

**Reply by sergeynilov** [reply]
Yes, that is just what I need, but pasting your example code :
```
                        Forms\Components\TextInput::make('amount')->numeric()
                            ->prefix(fn() => new HtmlString('<img src="https://flagsapi.com/KE/flat/32.png">')),
```

 I see raw html : https://img001.prntscr.com/file/img001/L3JNz1reQ0y4m3m-tRLLKg.png

**Reply by leandrocfe** [reply]
?
```php
Forms\Components\TextInput::make('amount')
    ->numeric()
    ->prefix(new \Illuminate\Support\HtmlString('<img src="https://flagsapi.com/KE/flat/32.png" />'))
```

**Reply by sergeynilov** [reply]
Yes, your code shows raw html, not rendered. I upgrated composer.json and have now :

```
        "laravel/framework": "^10.31.0",
        "bezhansalleh/filament-exceptions": "^2.0.1",
        "filament/filament": "^3.0-stable",
        "filament/widgets": "^3.0-stable",

```
Can I set some options in resource ?

**Reply by leandrocfe** [reply]
Could you share the whole code or the repo on github please?

**Reply by sergeynilov** [reply]
I uploaded a project at https://github.com/sergeynilov/quizzes
Please, read readme.md.
Thank you very much!

**Reply by leandrocfe** [reply]
I'll try it

**Reply by leandrocfe** [reply]
![image](https://github.com/filamentphp/filament/assets/3833889/3510f3e0-542b-42f2-b14a-0d3e782f8b4a)

**Reply by sergeynilov** [reply]
Very strange, I got raw text. In my os I have : 
```
PHP Version 8.2.11
Linux master-at-home 6.2.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Oct  6 10:23:26 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

```
Can it be some php config options or pluging ?

**Reply by leandrocfe** [reply]
try this
```php
Forms\Components\TextInput::make('amount')->label('Check Icon 987')
    ->prefix(view('image'))
```

```html
<!-- resources/views/image.blade.php -->
<img src="https://flagsapi.com/KE/flat/32.png" />
```

**Reply by sergeynilov** [reply]
Anyway I see a raw html. Looks like on my side rendered as {{ $code }}, but on your as {!! $code !!} on your side ...

**Reply by leandrocfe** [reply]
It should work. Probably an issue in your setup

**Reply by sergeynilov** [reply]
What need I to check in my setup ?

**Reply by leandrocfe** [reply]
no idea...

**Reply by emmadesilva** [reply]
The prefix/suffix helpers do not seem to support HtmlStrings, as they escape the HTML within.

**Comment by sergeynilov** [comment]
From some moment prefix with language icon rendered normally as images.
Not sure why  so, but last time I created a custom style, as I read 
https://filamentphp.com/docs/3.x/support/style-customization

with commands :

`php artisan make:filament-theme`

and 

`php artisan make:filament-theme app`

Myabe that isfluenced my app - I do not know actually
